ext/TenetKrylovKitExt.jl

```diff
@@ -25,9 +25,31 @@ function eigsolve_prehook_tensor_reshape(A::Tensor, left_inds, right_inds)
     return Amat, left_sizes, right_sizes
 end
 
-function KrylovKit.eigselector(A::Tensor, T::Type; left_inds=Symbol[], right_inds=Symbol[], kwargs...)
-    Amat, _, _ = eigsolve_prehook_tensor_reshape(A, left_inds, right_inds)
-    return KrylovKit.eigselector(Amat, T; kwargs...)
+function eigsolve_prehook_tensor_reshape(A::Tensor, x₀::Tensor, left_inds, right_inds)
+    left_inds, right_inds = Tenet.factorinds(A, left_inds, right_inds)
+
+    # Determine the left and right indices
+    left_sizes = size.((A,), left_inds)
+    prod_left_sizes = prod(left_sizes)
+
+    inds(x₀) != left_inds && throw(
+        ArgumentError(
+            "The initial guess must have the same left indices as the tensor, but got $(inds(x₀)) and $left_inds."
+        ),
+    )
+    prod(size.((x₀,), left_inds)) != prod_left_sizes && throw(
+        ArgumentError(
+            "The initial guess must have the same size as the left indices, but got sizes $prod_x₀_sizes and $prod_left_sizes.",
+        ),
+    )
+
+    Amat, left_sizes, right_sizes = eigsolve_prehook_tensor_reshape(A, left_inds, right_inds)
```

Comment:
Move this up so you can reuse `left_sizes`.

Reply:
Okay, done.